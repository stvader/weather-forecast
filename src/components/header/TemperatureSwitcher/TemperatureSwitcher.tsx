import React from 'react';
import { Switch, Typography } from 'antd';
import { CELSIUS_SIGN, FAHRENHEIT_SIGN } from '../../../constants/temperatureUnitConstants';

import { bc, bcAnt } from '../../../utils/bem-cn';

const { Text } = Typography;

interface ITemperatureSwitcherView {
  isChecked: boolean;
  handleChange: any;
}

const b = bc('tempereture-switcher');
const bAnt = bcAnt(b);

const TemperatureSwitcher: React.FC<ITemperatureSwitcherView> = ({
  isChecked,
  handleChange,
}: ITemperatureSwitcherView) => (
  <div className={b('wrapper')}>
    <Text strong className={bAnt('temp-unit-note')}>
      {FAHRENHEIT_SIGN}
    </Text>
    <Switch checked={isChecked} defaultChecked onChange={handleChange} />
    <Text strong className={bAnt(['temp-unit-note', 'temp-unit-note-colored'])}>
      {CELSIUS_SIGN}
    </Text>
  </div>
);

export default TemperatureSwitcher;
