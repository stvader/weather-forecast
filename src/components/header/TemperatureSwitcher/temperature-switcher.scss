@import '../../../constants/colors.scss';

.tempereture-switcher__wrapper {
  text-align: right;
}

.tempereture-switcher__temp-unit-note {
  display: inline-block;
  margin: 0 10px;
}

.tempereture-switcher__temp-unit-note-colored {
  color: $blue-base;
}
