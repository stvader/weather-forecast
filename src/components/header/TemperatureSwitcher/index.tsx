import React from 'react';
import { connect } from 'react-redux';

import { IAppState } from '../../../reducers/types';
import { switchTemperatureUnit } from '../../../actions';
import { selectTemperatureSwitcher } from '../../../selectors';
import { CELSIUS, TempUnit } from '../../../constants/temperatureUnitConstants';

import TemperatureSwitcher from './TemperatureSwitcher';

import './temperature-switcher.scss';

interface ITemperatureSwitcher {
  temeratureUnit: TempUnit;
  handleChange: () => void;
}

const TemperatureSwitcherContainer: React.FC<ITemperatureSwitcher> = ({
  temeratureUnit,
  handleChange,
}: ITemperatureSwitcher) => {
  const isChecked = temeratureUnit === CELSIUS;

  return <TemperatureSwitcher isChecked={isChecked} handleChange={handleChange} />;
};

const mapStateToProps = (state: IAppState) => ({
  temeratureUnit: selectTemperatureSwitcher(state),
});

const mapDispatchToProps = (dispatch: any) => ({
  handleChange: () => {
    dispatch(switchTemperatureUnit());
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(TemperatureSwitcherContainer);
