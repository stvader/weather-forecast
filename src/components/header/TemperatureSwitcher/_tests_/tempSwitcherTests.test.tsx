import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Switch, Typography } from 'antd';
import TemperatureSwitcher from '../TemperatureSwitcher';

const { Text } = Typography;

configure({ adapter: new Adapter() });

const mockHandleChange = jest.fn();

describe('TemperatureSwitcher', () => {
  it('render TemperatureSwitcher with C switch', () => {
    const component = shallow(<TemperatureSwitcher isChecked handleChange={mockHandleChange} />);

    const element = component.find('.tempereture-switcher-container');
    const switcher = element.find(Switch);

    expect(component).toMatchSnapshot();
    expect(element).toBeDefined();
    expect(element.children()).toHaveLength(3);
    expect(element.find(Text)).toHaveLength(2);
    expect(switcher).toHaveLength(1);
    expect(switcher.prop('checked')).toBe(true);
    expect(switcher.prop('defaultChecked')).toBe(true);
  });

  it('render TemperatureSwitcher with K switch', () => {
    const component = shallow(
      <TemperatureSwitcher isChecked={false} handleChange={mockHandleChange} />,
    );

    const element = component.find('.tempereture-switcher-container');

    expect(component).toMatchSnapshot();
    expect(element.find(Switch).prop('checked')).toBe(false);
  });
});
