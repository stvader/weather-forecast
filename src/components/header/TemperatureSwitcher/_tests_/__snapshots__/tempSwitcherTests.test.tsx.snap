// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemperatureSwitcher render TemperatureSwitcher with C switch 1`] = `ShallowWrapper {}`;

exports[`TemperatureSwitcher render TemperatureSwitcher with K switch 1`] = `ShallowWrapper {}`;
